contracts/gasprice_minimum/gasprice_minimum.go

```diff
@@ -80,27 +78,43 @@ const (
 )
 
 var (
-	gasPriceMinimumABI, _            = abi.JSON(strings.NewReader(gasPriceMinimumABIString))
 	FallbackGasPriceMinimum *big.Int = big.NewInt(0) // gas price minimum to return if unable to fetch from contract
 	suggestionMultiplier    *big.Int = big.NewInt(5) // The multiplier that we apply to the minimum when suggesting gas price
 )
 
-func GetGasPriceSuggestion(currency *common.Address, header *types.Header, state vm.StateDB) (*big.Int, error) {
-	gasPriceMinimum, err := GetGasPriceMinimum(currency, header, state)
+var (
+	getGasPriceMinimumMethod    *contracts.BoundMethod
+	updateGasPriceMinimumMethod *contracts.BoundMethod
+)
+
+func init() {
+	gasPriceMinimumABI, err := abi.JSON(strings.NewReader(gasPriceMinimumABIString))
+	if err != nil {
+		panic(err)
+	}
+
+	getGasPriceMinimumMethod = contracts.NewRegisteredContractMethod(params.GasPriceMinimumRegistryId, &gasPriceMinimumABI, "getGasPriceMinimum", params.MaxGasForGetGasPriceMinimum)
+	updateGasPriceMinimumMethod = contracts.NewRegisteredContractMethod(params.GasPriceMinimumRegistryId, &gasPriceMinimumABI, "updateGasPriceMinimum", params.MaxGasForUpdateGasPriceMinimum)
+}
+
+func GetGasPriceSuggestion(vmRunner vm.EVMRunner, currency *common.Address) (*big.Int, error) {
+	gasPriceMinimum, err := GetGasPriceMinimum(vmRunner, currency)
 	return new(big.Int).Mul(gasPriceMinimum, suggestionMultiplier), err
 }
 
-func GetGasPriceMinimum(currency *common.Address, header *types.Header, state vm.StateDB) (*big.Int, error) {
+func GetGasPriceMinimum(vmRunner vm.EVMRunner, currency *common.Address) (*big.Int, error) {
 	var currencyAddress common.Address
 	var err error
 
+	// TODO this is not ideal, we fail on vmRunner == nil
```

Comment:
Why would vmRunner be nil?  And why is this necessary here but not elsewhere?  Wouldn't all the contract methods go wrong if it were nil?  Is it because of `tracers_test.go`?  If so, I would rather create some stub struct that satisfies the `VMRunner` interface for that test so that we don't need to have special handling here for what only happens in that one test.

Reply:
Actually, I know see you already have a `fakeEVMRunner`.  Couldn't it be used in tracers_test.go too, and if so then would we be able to remove this here?

Reply:
using celoMock runner now